src/commands/createFunction/durableSteps/netherite/EventHubsNamespaceCreateStep.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { EHNamespace, EventHubManagementClient } from '@azure/arm-eventhub';
+import { AzureWizardExecuteStep, ISubscriptionContext, nonNullValue } from '@microsoft/vscode-azext-utils';
+import { Progress } from 'vscode';
+import { ext } from '../../../../extensionVariables';
+import { localize } from '../../../../localize';
+import { createEventHubClient } from '../../../../utils/azureClients';
+import { IEventHubsConnectionWizardContext } from '../../../appSettings/IEventHubsConnectionWizardContext';
+
+export class EventHubsNamespaceCreateStep<T extends IEventHubsConnectionWizardContext> extends AzureWizardExecuteStep<T> {
+    public priority: number = 200;
+
+    public async execute(context: T, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+        const rgName: string = nonNullValue(context.resourceGroup?.name);
+        const newNamespaceName: string = nonNullValue(context.newEventHubsNamespaceName);
+        const creating: string = localize('creatingEventHubsNamespace', 'Creating new event hubs namespace "{0}"...', newNamespaceName);
+        ext.outputChannel.appendLog(creating);
+        progress.report({ message: creating });
+
+        const client: EventHubManagementClient = await createEventHubClient(<T & ISubscriptionContext>context);
+        const defaultParams: EHNamespace = {
+            location: nonNullValue(context.resourceGroup?.location),
```

Comment:
Discussed offline: Use `LocationListStep.getLocation` here after invoking `LocationListStep.setLocation` based off the resource group's location.